addons/l10n_sa_edi/models/account_move.py

```diff
@@ -168,8 +175,21 @@ def _l10n_sa_log_results(self, xml_content, response_data=None, error=False):
             xml_content)
         bootstrap_cls, title, content = ("success", _("Invoice Successfully Submitted to ZATCA"),
                                          "" if (not error or not response_data) else response_data)
+        attachment = False
         if error:
+            xml_filename = self.env['account.edi.xml.ubl_21.zatca']._export_invoice_filename(self)
+            xml_filename = xml_filename[:-4] + '-rejected.xml'
+            attachment = self.env['ir.attachment'].with_context(no_new_invoice=True).create({
```

Comment:
The context should be on the `message_post`, as I have the message (not updated from the attachment)

Reply:
Nvm :)